src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -90,7 +99,12 @@ void M5(S<T> s5)
         M2(ref s5.F1);
     }
 }";
-            var comp = CreateCompilation(sourceA, parseOptions: TestOptions.Regular10);
+            var mscorlibRefWithRefFields = GetMscorlibRefWithoutSharingCachedSymbols();
+
+            // Note: we use skipUsesIsNullable and skipExtraValidation so that nobody pulls
```

Comment:
I'm confused why this is necessary. I wouldn't expect anything to be pulling on the compilation or it's references until the compilation is actually used? Can you clarify why we need to do this for this particular flag, but haven't needed to for any of the other `RuntimeSupports` bypass flags?

Reply:
We don't need this for this particular flag, but we need it for certain flags such as numeric IntPtr (pattern that I'm following). 
I'll remove but we should have a discussion at some point on a universal test hook, so that we don't have to special-case every new runtime feature flag.

Reply:
I opened an issue ([https://github.com/dotnet/roslyn/issues/63223](https://github.com/dotnet/roslyn/issues/63223)) to sit down and design a general/re-usable CreateCompilation API supporting what we need, including a note on `skipUsesIsNullable` and `skipExtraValidation` since some features need those but some don't.